packages/workbox-build/src/entry-points/generate-sw-no-fs.js

```diff
@@ -0,0 +1,32 @@
+const GenerateSWNoFSOptions = require('./options/generate-sw-no-fs-options');
+const getFileManifestEntries = require('../lib/get-file-manifest-entries');
+const populateSWTemplate = require('../lib/populate-sw-template');
+
+/**
+ * This method generates a service worker based on the configuration options
+ * provided without reading or writing to the disk.
+ *
+ * Both the source template (in a format parsable by
+ * [`lodash.template`](https://lodash.com/docs/4.17.4#template)) and the
+ * populated template are strings.
+ *
+ * The ideal use case for this method is when you have a larger build system
+ * that has its own way of interacting with the filesystem (such as a Webpack).
+ *
+ * @param {Object} input
+ * @return {Promise<String>} A populated service worker template, based on the
+ * other configuration options provided.
+ *
+ * @memberof module:workbox-build
+ */
+async function generateSWNoFS(input) {
```

Comment:
This is one of the new interfaces, catered to @goldhand's use case in the Webpack plugin where both the template used as input and the final output be passed as strings, rather than use the filesystem.

The `generateSWNoFS()` reflects that it is similar to `generateSW()`, without using the filesystem for anything, but I'd imagine that name should change. (Any ideas?)

Apologies re: the lack of JSDocs right now, but you can see usage within the corresponding end to end test.

Reply:
Sure, I'll go with `generateSWString()`.